### DISCUSSION TITLE
micrOS & Wemos/Lolin boards (esp8266/ESP32/ESP32-S2)

### DISCUSSION THREAD

**Discussion by durapensa** [BxNxM/micrOS#10] [General]
I went looking for a local-network IoT firmware for making useful gadgets for friends (i.e. easy smartphone control is a requirement) and found micrOS.

My own gadgets run CircuitPython, MicroPython & Tasmota (with Home Assistant). I've experimented with a few other 'IoT frameworks' but always chose to remain with these three, using a local MQTT server & Adafruit IO. I think that micrOS is what I've been looking for!

So far, I flashed  a Wemos/Lolin D1 mini (esp8266) with v1.2.2-0 and poked around in the shell.

Next steps are to create a LogicalPin lookup table for the D1 mini (based on a cut-down `LP_esp8266.py`) and get a NeoPixel example working.

I worked up the CircuitPython board definition for the Lolin S2 Mini (a new $4  ESP32-S2 board & similar to the TinyS2 & FeatherS2 by the Unexpected Maker), and have some LOLIN32 boards coming in the mail (ESP32 & similar to the Unexpected Maker's TinyPICO but with LiPo charging). I'm happy to experiment & provide feedback on micrOS & ESP32-S2 boards if you plan on supporting ESP32-S2.

**Comment by BxNxM** [comment]
Hi,

You can add "custom" micropython images under https://github.com/BxNxM/micrOS/tree/master/micrOS/micropython (reporoot/micrOS/micropython) folder on your machine.
Then the framework will list that image as well on the DevToolKit GUI, and theoretically the micrOS resources are able to run on any micropython version, supporting wifi and irqs. 

So you can have experiment with that. :)

**Reply by BxNxM** [reply]
esp32-s2 board support is done
